# Project 04 -- Anthony Micco

**TA Help:** N/A
    
**Collaboration:** N/A

## Question 1

In [16]:
# navigating to the restaurant directory and printing the second column of orders.csv

In [17]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
head /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," '{print $2}'

customer_id
92PEE24
QS68UD8
MB7VY5F
KDJ951Y
BAL0RVT
U263OCD
I9DNSMJ
I9DNSMJ
QYXXJCF


In [18]:
#finding all the .csv files located in the restaurant directory

In [19]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
ls /anvil/projects/tdm/data/restaurant/ | awk '/\.csv$/'

SampleSubmission (1).csv
orders.csv
test_customers.csv
test_full.csv
test_locations.csv
train_customers.csv
train_full.csv
train_locations.csv
vendors.csv


In [20]:
#printing the second column of the first five lines of tracks.csv 

In [27]:
%%bash
cd /anvil/projects/tdm/data/amazon/
cat /anvil/projects/tdm/data/amazon/tracks.csv | awk -F"," '{print $2}' | head -n 5

1.4) The default field separator when using the awk command is simply whitespace. If we want to use a different delimiter we can use the -F command with a specific delimiter such as -F"," to use commas as a delimiter. 

## Question 2

In [28]:
#finding the total item count sold in orders.csv 

In [30]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
head /anvil/projects/tdm/data/restaurant/orders.csv

akeed_order_id,customer_id,item_count,grand_total,payment_mode,promo_code,vendor_discount_amount,promo_code_discount_percentage,is_favorite,is_rated,vendor_rating,driver_rating,deliverydistance,preparationtime,delivery_time,order_accepted_time,driver_accepted_time,ready_for_pickup_time,picked_up_time,delivered_time,delivery_date,vendor_id,created_at,LOCATION_NUMBER,LOCATION_TYPE,CID X LOC_NUM X VENDOR
163238.0,92PEE24,1.0,7.6,2,,0.0,,,No,,0.0,0.0,,,,,,,,2019-07-31 05:30:00,105,2019-08-01 05:30:16,0,,92PEE24 X 0 X 105
163240.0,QS68UD8,1.0,8.7,1,,0.0,,,No,,0.0,0.0,,,,,,,,2019-07-31 05:30:00,294,2019-08-01 05:31:10,0,Work,QS68UD8 X 0 X 294
163241.0,MB7VY5F,2.0,14.4,1,,0.0,,,No,,0.0,0.0,,,,,,,,2019-07-31 05:30:00,83,2019-08-01 05:31:33,0,,MB7VY5F X 0 X 83
163244.0,KDJ951Y,1.0,7.1,1,,0.0,,,No,,0.0,0.0,,2019-08-01 04:03:00,,,,,,2019-07-31 05:30:00,90,2019-08-01 05:34:54,0,Home,KDJ951Y X 0 X 90
163245.0,BAL0RVT,4.0,27.2,1,,0.0,,,No,,0.0,0.0,,,,,,,,2019-07-31 05:30:00,83,2019-08-01 05:35:51,0,

In [37]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," 'BEGIN{count=0} {count+=$3} END{print "The total items sold is: " count}'

The total items sold is: 308801


In [38]:
#Finding the total revenue in orders.csv

In [41]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," 'BEGIN{revenue=0} {if ($7 == 0.0) {revenue+=$4}} END{print "The total revenue is: " revenue}'

The total revenue is: 2.07822e+06


In [42]:
#modifying the print component to not use scientific notation

In [46]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," 'BEGIN{revenue=0} {if ($7 == 0.0) {revenue+=$4}} END{printf "The total revenue is: " "%.2f\n", revenue}'

The total revenue is: 2078216.30


In [47]:
#Finding the total count of all orders greater than $100

In [49]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," 'BEGIN{count=0} {if ($4 > 100.0) {count+=1}} END{print "The # of orders over $100 is: " count}'

The # of orders over $100 is: 226


In [50]:
#Finding the most frequently ordered customer

In [54]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," '{print $2}' | sort | uniq -c | sort -n | tail

    107 HX76ORD
    109 BUIAFFT
    113 TVSPG2H
    114 KXXRD9J
    115 2JJAO46
    119 B28LJKM
    128 HJFTTGW
    146 VDEJEMP
    151 TL7Z2DM
    262 XW90EAP


In [55]:
#Finding the total amount of money the customer spent

In [56]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," 'BEGIN{total=0} {if ($2 == "XW90EAP") {total+=$4}} END{printf "The customer spent: " "%.2f\n", total}'

The customer spent: 3830.60


2.6) The customer with the most orders is customer **XW90EAP** \
2.7) This customer spent **$3830.60**

## Question 3

In [57]:
# finding the total number of orders for each gender group

In [60]:
%%bash
cd /anvil/projects/tdm/data/restaurant/
awk -F"," 'NR > 1 {count[$2]++} END {for (g in count) print g, count[g]}' /anvil/projects/tdm/data/restaurant/test_full.csv

 520000
Male 1062300
Female 89700


In [61]:
# finding how many orders were placed at a customer's favorite restaurant

In [69]:
%%bash
cd /anvil/projects/tdm/data/restaurant/
awk -F"," 'NR > 1 {if ($9) {count++}} END {print count}' /anvil/projects/tdm/data/restaurant/orders.csv

100108


In [70]:
#finding which vendor was marked as a favorite by the most number of customers

In [15]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," '{if ($9) print "Vendor: " $22}' | sort | uniq -c | sort -n | tail

   1578 Vendor: 191
   1622 Vendor: 356
   1928 Vendor: 189
   2559 Vendor: 846
   2833 Vendor: 86
   3280 Vendor: 386
   4480 Vendor: 105
   4961 Vendor: 84
   5006 Vendor: 113
   5093 Vendor: 79


3.3) **100108** orders were placed from a customer's favorite vendor. \
3.4) The vendor marked favorite by the most customers is **vendor id 79** with 5093 customers.

## Question 4

In [76]:
# finding which vendor has the highest rating ordered from by gender

In [78]:
%%bash 
head -n 1 /anvil/projects/tdm/data/restaurant/test_full.csv

customer_id,gender,status_x,verified_x,created_at_x,updated_at_x,location_number,location_type,latitude_x,longitude_x,id,authentication_id,latitude_y,longitude_y,vendor_category_en,vendor_category_id,delivery_charge,serving_distance,is_open,OpeningTime,OpeningTime2,prepration_time,commission,is_akeed_delivering,discount_percentage,status_y,verified_y,rank,language,vendor_rating,sunday_from_time1,sunday_to_time1,sunday_from_time2,sunday_to_time2,monday_from_time1,monday_to_time1,monday_from_time2,monday_to_time2,tuesday_from_time1,tuesday_to_time1,tuesday_from_time2,tuesday_to_time2,wednesday_from_time1,wednesday_to_time1,wednesday_from_time2,wednesday_to_time2,thursday_from_time1,thursday_to_time1,thursday_from_time2,thursday_to_time2,friday_from_time1,friday_to_time1,friday_from_time2,friday_to_time2,saturday_from_time1,saturday_to_time1,saturday_from_time2,saturday_to_time2,primary_tags,open_close_flags,vendor_tag,vendor_tag_name,one_click_vendor,country_id,city_id,created_at_y,updat

In [96]:
%%bash
cd /anvil/projects/tdm/data/restaurant/
awk -F"," 'NR > 1 {gender = $2; if ($30 > max[gender]) {max[gender] = $30}} END {for (g in max) {print g, max[g]}}' test_full.csv

 4.8
Male 4.8
Female 4.8


In [95]:
#finding the average vendor rating by gender

In [109]:
%%bash
cd /anvil/projects/tdm/data/restaurant/
awk -F"," 'NR > 1 {gender = $2; rating=$30; sum[gender] += rating; count[gender]++;} END {for (g in sum) { avg = sum[g] / count[g];  printf "%s %.2f\n", g, avg}}' test_full.csv

 4.01
Male 4.35
Female 4.35


In [1]:
#finding which vendor offered the most discounts

In [17]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," '{if ($7 >0) print "Vendor: " $22}' | sort | uniq -c | sort -n | tail

      1 Vendor: 66
      1 Vendor: vendor_id
     13 Vendor: 85
    183 Vendor: 148


In [27]:
#finding the total number of orders for vendor 148

In [26]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," 'BEGIN{total=0} {if ($22 == "148") {total++}} END{print total}'

1078


In [28]:
 #finding the percentage with a discount for vendor 148

In [34]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
cat /anvil/projects/tdm/data/restaurant/orders.csv | awk -F"," 'BEGIN{total=0} {if ($22 == "148" && $7 >0) {total++}} END{printf "%.2f\n", total / 1078 * 100}'

16.98


In [35]:
#Finding the vendor that had the most orders delivered to work

In [7]:
%%bash 
cd /anvil/projects/tdm/data/restaurant/
awk -F"," 'NR > 1 {if ($25== "Work") {count[$22]++}} END{max = 0; for(v in count){if(count[v]>max) max=count[v]; vendor=v;} print vendor, max}' /anvil/projects/tdm/data/restaurant/orders.csv

237 989


4.2) The highest rating for a vendor by gender is: male: 4.8 and female: 4.8  \
4.3) The average rating for each vendor by gender is: male: 4.35 and female: 4.35\
4.4) Vendor ID 148 offered the most discounts \
4.5) Vendor 148 received 1078 orders and only 16.98 % had discounts \
4.6) Vendor 237 had the most orders delivered to the location_type "work"

## Question 5

In [12]:
#viewing the head of train_locations.csv

In [14]:
%%bash 
head -n 10 /anvil/projects/tdm/data/restaurant/train_locations.csv

customer_id,location_number,location_type,latitude,longitude
02SFNJH,0,NA,1.6823915384615757,-78.7897371818186
02SFNJH,1,NA,1.6791365450962032,0.7668233053886399
02SFNJH,2,NA,-0.4986483973653315,0.6612414365454651
RU43CXC,0,Home,0.10085325202709132,0.43816546553915037
BDFBPRD,0,NA,2.523125243914389,0.7334637242284848
WMGKW6W,0,NA,12.66341321723571,-1.429919200187407
NDYLK9A,0,NA,-0.20327980769232123,-78.56688511363677
NDYLK9A,1,NA,-0.20437114344229615,0.43559274875681514
PB2B28D,0,Home,2.5756051968780787,0.7082701813272576


In [ ]:
# calculating the top 5 customers and recreating the formatted output

In [21]:
%%bash
cd /anvil/projects/tdm/data/restaurant/
awk -F, 'NR > 1 {count[$1]++; total++} END {for (c in count) print count[c], c; print "TOTAL: ", total}' train_locations.csv | sort -k1,1nr | awk 'BEGIN { print "**************************************************"; print "*              Our Best Customers                *"; print "**************************************************"}$1 != "TOTAL" && NR <= 5 { printf "%-2s %s\n", $1, $2 }$1 == "TOTAL" {print "**************************************************"; printf "* Total lines processed: %-23s *\n", $2; print "**************************************************"}'

**************************************************
*              Our Best Customers                *
**************************************************
30 4Y0K8NQ
26 NETYQ1C
20 5EP9F62
19 4UZKXW2
19 YYKZN0F


## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.